weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallPage/ValueViewNumberTimestamp.tsx

```diff
@@ -7,18 +7,32 @@ const JAN_1_2100_S = 4_102_444_800;
 const JAN_1_2000_MS = 1000 * JAN_1_2000_S;
 const JAN_1_2100_MS = 1000 * JAN_1_2100_S;
 
-// TODO: This is only looking at value, but we could also consider the value name, e.g. "created".
+// Regex pattern to match likely timestamp field names
+const TIMESTAMP_PATTERN =
+  /^(?:.*_)?(created|started|ended|updated|finished|duration|time|timestamp)(?:_(?:at|ms))?$/i;
 
-export const isProbablyTimestampMs = (value: number) => {
-  return JAN_1_2000_MS <= value && value <= JAN_1_2100_MS;
+export const likelyTimestampName = (field?: string) => {
+  return field ? TIMESTAMP_PATTERN.test(field) : false;
 };
 
-export const isProbablyTimestampSec = (value: number) => {
-  return JAN_1_2000_S <= value && value <= JAN_1_2100_S;
+export const isProbablyTimestampMs = (value: number, field?: string) => {
+  return (
+    likelyTimestampName(field) &&
+    JAN_1_2000_MS <= value &&
+    value <= JAN_1_2100_MS
+  );
 };
 
-export const isProbablyTimestamp = (value: number) => {
-  return isProbablyTimestampMs(value) || isProbablyTimestampSec(value);
+export const isProbablyTimestampSec = (value: number, field?: string) => {
+  return (
+    likelyTimestampName(field) && JAN_1_2000_S <= value && value <= JAN_1_2100_S
+  );
+};
+
+export const isProbablyTimestamp = (value: number, field?: string) => {
```

Comment:
It seems like if field is not given the function will alway return false so not sure how useful it is to make it an optional arg instead of a required one?